File: TheWorld/src/TheWorld/Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using TheWorld.Models;
using Microsoft.Extensions.Logging;
using TheWorld.ViewModels;
using Microsoft.AspNetCore.Authorization;

// For more information on enabling MVC for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860

namespace TheWorld.Controllers.Api
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller
    {
        IWorldRepository Repository;
        ILogger Logging;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logging)
        {
            Repository = repository;
            Logging = logging;
        }


        //[HttpGet("api/trips")]
        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                var results = Repository.GetTripsByUserName(User.Identity.Name);

                return Ok(AutoMapper.Mapper.Map<IEnumerable<TripViewModel>>(results));
            }
            catch (Exception ex)
            {
                Logging.LogCritical($"Failed to get Trips{ex}");
                return BadRequest("Error occurred");
            }

        }

        //[HttpPost("api/trips")]
        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                var newTrip = AutoMapper.Mapper.Map<Trip>(theTrip);

                newTrip.UserName = User.Identity.Name;

                Repository.AddTrip(newTrip);

                if (await Repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{newTrip.Name}", AutoMapper.Mapper.Map<TripViewModel>(newTrip));
                }

                return BadRequest("Failed to save changes to db");
            }
            return BadRequest("Bad data");
        }
    }
}

File: TheWorld/src/TheWorld/wwwroot/js/App/tripsController.js
(function () {
    "use strict";
    angular.module("app-trips").controller("tripsController", tripsController);

    function tripsController($http) {

        var vm = this;

        vm.errorMessage = "";

        vm.isBusy = true;

        vm.trips = [];

        vm.newTrip = {};

        $http.get("/api/trips")
        .then(function (response) {
            //success
            angular.copy(response.data, vm.trips);
        },
        function (error) {
            //fail
            vm.errorMessage = "Failed to load data. " + error;
        }).finally(function () {
            vm.isBusy = false;
        });
        vm.addTrip = function () {
            vm.isBusy = true;
            vm.errorMessage = "";
            $http.post("/api/trips", vm.newTrip)
            .then(function (response) {
                //success
                vm.trips.push(response.data);
                //clear form
                vm.newTrip = {};
            }, function (error) {
                //fail
                vm.errorMessage = "Failed to save new trip. " + error;
            }).finally(function () {
                vm.isBusy = false;
                
            });

        };
    };


})();